filebeat/harvester/reader/multiline.go

```diff
@@ -112,8 +114,18 @@ func (mlr *Multiline) Next() (Message, error) {
 	return mlr.state(mlr)
 }
 
+func (r *Multiline) Close() error {
+	close(r.done)
+	return r.reader.Close()
+}
+
 func (mlr *Multiline) readFirst() (Message, error) {
 	for {
+		select {
+		case <-mlr.done:
+			return Message{}, ErrReaderStopped
+		default:
+		}
```

Comment:
we need to test the done channel here, with us closing the underlying reader anyways? We need the done channel at all?